heartbeat/corosync-qnetd

```diff
@@ -0,0 +1,347 @@
+    <shortdesc lang="en">corosync-qnetd nss db directory</shortdesc>
+    <content type="string" default="${OCF_RESKEY_nss_db_dir_default}" />
+</parameter>
+
+<parameter name="pid">
+    <longdesc lang="en">
+        Location of the corosync-qnetd pid/lock
+    </longdesc>
+    <shortdesc lang="en">corosync-qnetd pid file</shortdesc>
+    <content type="string" default="${OCF_RESKEY_pid_default}" />
+</parameter>
+
+<parameter name="port">
+    <longdesc lang="en">
+        TCP port to listen on. Default port is 5403.
+    </longdesc>
+    <shortdesc lang="en">TCP portto listen on</shortdesc>
+    <content type="string" default="${OCF_RESKEY_port_default}" />
+</parameter>
+
+<parameter name="qnetd_opts">
+    <longdesc lang="en">
+        Additional options for corosync-qnetd binary. "-4" for example.
+    </longdesc>
+    <shortdesc lang="en">corosync-qnetd extra options</shortdesc>
+    <content type="string" default="${OCF_RESKEY_qnetd_opts_default}" />
+</parameter>
+
+<parameter name="qnetd_tool_binary">
+    <longdesc lang="en">
+        The absolute path to the corosync-qnetd-tool for monitoring with OCF_CHECK_LEVEL greater zero.
+    </longdesc>
+    <shortdesc lang="en">The absolute path to the corosync-qnetd-tool binary</shortdesc>
+    <content type="string" default="${OCF_RESKEY_qnetd_tool_binary_default}" />
+</parameter>
+
+<parameter name="user">
+    <longdesc lang="en">User running corosync-qnetd</longdesc>
+    <shortdesc lang="en">corosync-qnetd user</shortdesc>
+    <content type="string" default="${OCF_RESKEY_user_default}" />
+</parameter>
+
+</parameters>
+
+<actions>
+<action name="start" timeout="20s" />
+<action name="stop" timeout="20s" />
+<action name="status" timeout="20s" />
+<action name="monitor" depth="0" timeout="20s" interval="10s" start-delay="10s" />
+<action name="validate-all" timeout="20s" />
+<action name="meta-data" timeout="20s" />
+</actions>
+</resource-agent>
+END
+}
+
+corosync_qnetd_status() {
+    ocf_pidfile_status "${OCF_RESKEY_pid}" > /dev/null 2>&1
+    case "$?" in
+        0)
+            rc="$OCF_SUCCESS"
+            ;;
+        1|2)
+            rc="$OCF_NOT_RUNNING"
+            ;;
+        *)
+            rc="$OCF_ERR_GENERIC"
+            ;;
+    esac
+
+    return "$rc"
+}
+
+corosync_qnetd_start() {
+    corosync_qnetd_validate_all
+    rc="$?"
+
+    if [ "$rc" -ne 0 ]; then
+        return "$rc"
+    fi
+
+    # if resource is already running,no need to continue code after this.
+    if corosync_qnetd_status; then
+        ocf_log info "corosync-qnetd is already running"
+        return "${OCF_SUCCESS}"
+    fi
+
+    pid_dir=$(dirname "${OCF_RESKEY_pid}")
+    sock_dir=$(dirname "${OCF_RESKEY_ipc_sock}")
+
+    for d in "$pid_dir" "$sock_dir";do
+        if [ ! -d "$d" ];then
+            mkdir -p "$d"
+            chmod 0770 "$d"
+            chown "${OCF_RESKEY_user}:${OCF_RESKEY_user}" "$d"
+        fi
+    done
+
+    if [ -n "${OCF_RESKEY_nss_db_dir}" ];then
```

Comment:
It's probably better to set `params=""` and do `params="$params ..."` in the if's here.

Reply:
I've moved everything to params="$params ...." (even the "hardcoded" sock/pid) so resulting ocf_run stanza is much shorter.